src/malva/java/math/BigDecimalTest.java

```diff
@@ -0,0 +1,24 @@
+package malva.java.math;
+
+import java.math.BigDecimal;
+
+import malva.TestCase;
+
+public class BigDecimalTest extends TestCase {
+  public static void testStrippingZeros() {
+    // Array as { test, result }
+    final BigDecimal[][] testCases = {
+      {new BigDecimal("0.1234000"), new BigDecimal("0.1234")},
+      {new BigDecimal("0e2"), BigDecimal.ZERO},
+      {new BigDecimal("100000"), new BigDecimal("1e5")}
```

Comment:
Better to keep this as multiple assertEquals calls, this makes it easier to find out the offending test cases if a test fails (as line numbers will be included in the stack trace)

Reply:
Fixed